File: src/main/java/org/acg/user_services/repository/UserRepo.java
package org.acg.user_services.repository;

import org.acg.user_services.repository.entity.UserInfo;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepo extends CrudRepository<UserInfo, Long> {
    UserInfo findByEmail(String email);
}

File: src/main/kotlin/org/acg/user/repository/UserRepo.kt
package org.acg.user.repository;

import org.acg.user.repository.entity.UserInfo;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
interface UserRepo : CrudRepository<UserInfo, Long> {
    fun findByEmail(email: String): UserInfo?
}